packages/databricks-vscode/src/configuration/ConnectionManager.ts

```diff
@@ -98,125 +153,53 @@ export class ConnectionManager {
-            (!this._databricksWorkspace.isReposEnabled ||
-                !this._databricksWorkspace.isFilesInReposEnabled)
-        ) {
-            let message = "";
-            if (!this._databricksWorkspace.isReposEnabled) {
-                message =
-                    "Repos are not enabled for this workspace. Please enable it in the Databricks UI.";
-            } else if (!this._databricksWorkspace.isFilesInReposEnabled) {
-                message =
-                    "Files in Repos is not enabled for this workspace. Please enable it in the Databricks UI.";
-            }
-            NamedLogger.getOrCreate("Extension").error(message);
-            if (interactive) {
-                const result = await window.showWarningMessage(
-                    message,
-                    "Open Databricks UI"
-                );
-                if (result === "Open Databricks UI") {
-                    const host = await workspaceClient.apiClient.host;
-                    await env.openExternal(
-                        Uri.parse(
-                            host.toString() +
-                                "#setting/accounts/workspace-settings"
-                        )
-                    );
-                }
-            }
-        }
-
-        this._workspaceClient = workspaceClient;
-        this._projectConfigFile = projectConfigFile;
+        const databricksWorkspace = await DatabricksWorkspace.load(
+            authProvider.getWorkspaceClient(),
+            authProvider
+        );
+        this._databricksWorkspace = databricksWorkspace;
+        this._workspaceClient = authProvider.getWorkspaceClient();
 
-        if (projectConfigFile.clusterId) {
-            await this.attachCluster(projectConfigFile.clusterId, true);
-        } else {
-            this.updateCluster(undefined);
-        }
+        await this._databricksWorkspace.optionalEnableFilesInReposPopup(
+            this._workspaceClient
+        );
 
-        if (projectConfigFile.workspacePath) {
-            await this.attachSyncDestination(
-                new RemoteUri(projectConfigFile.workspacePath),
-                true
+        const workspacePath = await this.configModel.get("workspaceFsPath");
+        const remoteUri = workspacePath
+            ? new RemoteUri(workspacePath)
+            : undefined;
+        if (remoteUri !== undefined && !(await remoteUri.validate(this))) {
+            window.showErrorMessage(
+                `Invalid sync destination ${workspacePath}`
             );
-        } else {
-            this.updateSyncDestination(undefined);
+            this.detachSyncDestination();
         }
 
-        await this.createWsFsRootDirectory(workspaceClient);
+        await this.createWsFsRootDirectory(this._workspaceClient);
+        await this.setSyncDestinationMapper();
+        await this.setClusterManager();
+        await this.configModel.set("authParams", authProvider.toJSON());
+
         this.updateState("CONNECTED");
```

Comment:
If something fails before we reach this point (but after setting CONNECTING state) we might end up in a bad state (with hanging `waitForConnect` promises, not working `logout` method, and the UI not being updated properly). Was there a reason to remove the try catch fail-safe we had around `_login` before?

Reply:
I missed adding that to the `configureWorkspace` method later. The idea was that login is supposed to login and nothing else and fail in a weird state if that happens (because it never did reach any other state). And then the calling functions handle retry or logout.